src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -569,6 +576,11 @@ public PrivilegesEvaluatorResponse evaluate(
 
     }
 
+    public Boolean isServiceAccount(final User user) {
+        Map<String, String> userAttributesMap = user.getCustomAttributesMap();
+        return userAttributesMap != null && "true".equals(userAttributesMap.get("attr.internal.service"));
+    }
```

Comment:
This seems like a function that should be on the User class.

Reply:
makes sense! I will refactor it into it after some the testing work im doing.